src/loading-bar.js

```diff
@@ -227,37 +227,18 @@ angular.module('cfp.loadingBar', [])
       }
 
       /**
-       * Increments the loading bar by a random amount
-       * but slows down as it progresses
+       * Increments the loading bar by an amount but slows down as it progresses
        */
       function _inc() {
-        if (_status() >= 1) {
-          return;
-        }
+        var stat = _status(),
+            pct;
 
-        var rnd = 0;
-
-        // TODO: do this mathmatically instead of through conditions
-
-        var stat = _status();
-        if (stat >= 0 && stat < 0.25) {
-          // Start out between 3 - 6% increments
-          rnd = (Math.random() * (5 - 3 + 1) + 3) / 100;
-        } else if (stat >= 0.25 && stat < 0.65) {
-          // increment between 0 - 3%
-          rnd = (Math.random() * 3) / 100;
-        } else if (stat >= 0.65 && stat < 0.9) {
-          // increment between 0 - 2%
-          rnd = (Math.random() * 2) / 100;
-        } else if (stat >= 0.9 && stat < 0.99) {
-          // finally, increment it .5 %
-          rnd = 0.005;
-        } else {
-          // after 99%, don't increment:
-          rnd = 0;
+        if (stat >= 1) {
+            return;
         }
 
-        var pct = _status() + rnd;
+        pct = stat + (1 - Math.log(stat * 100) / Math.log(100)) / 3;
```

Comment:
Just out of curiosity to help my understanding, why are you subtracting from 1?  Why are you always dividing by ``Math.log(100)) / 3``? In fact, you are basically dividing by 6. How does that improve the animation?